arrabiata/src/constraints.rs

```diff
@@ -318,3 +319,69 @@ impl<Fp: PrimeField> InterpreterEnv for Env<Fp> {
         lambda
     }
 }
+
+impl<F: PrimeField> Env<F> {
+    /// Get all the constraints for the IVC circuit, only.
+    ///
+    /// The following gadgets are used in the IVC circuit:
+    /// - [Instruction::PoseidonNextRow] to verify the challenges and the public
+    /// IO
+    /// - [Instruction::EllipticCurveScaling] and
+    /// [Instruction::EllipticCurveAddition] to accumulate the commitments
+    // FIXME: the IVC circuit might not be complete, yet. For instance, we might
+    // need to accumulate the challenges and add a row to verify the output of
+    // the computation of the challenges.
+    // FIXME: add a test checking that whatever the value given in parameter of
+    // the gadget, the constraints are the same
+    pub fn get_all_constraints_for_ivc(&self) -> Vec<E<F>> {
+        // Copying the instance we got in parameter, and making it mutable to
+        // avoid modifying the original instance.
+        let mut env = self.clone();
+        // Resetting before running anything
+        env.reset();
+
+        let mut constraints = vec![];
+
+        // Poseidon constraints
+        // The constraints are the same for all the value given in parameter,
+        // therefore picking 0
+        interpreter::run_ivc(&mut env, Instruction::PoseidonNextRow(0));
```

Comment:
why do we need to run the interpreter to gather the constraints ?
Does the result of this function depends on self ?

Reply:
The steps of the IVC circuit is abstracted in the interpreter.
It depends on `self` only because there is a dependency on the Poseidon parameters. I would like to change that later (in particular, we need a setup).